pandas/tests/indexes/test_interval.py

```diff
@@ -327,171 +335,298 @@ def test_get_item(self):
+                                Interval(2, 3, closed=side)) == 1
+                        else:
+                            pytest.raises(KeyError, idx.get_loc,
+                                          Interval(*bound, closed=side))
+                    else:
+                        pytest.raises(KeyError, idx.get_loc,
+                                      Interval(*bound, closed=side))
 
-        # TODO: is this right?
-        # return indexers for wholly contained subintervals
-        target = IntervalIndex.from_breaks(np.linspace(0, 2, 5))
-        actual = self.index.get_indexer(target)
-        expected = np.array([0, 0, 1, 1], dtype='p')
-        tm.assert_numpy_array_equal(actual, expected)
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_get_loc_scalar_updated_behavior(self):
 
-        target = IntervalIndex.from_breaks([0, 0.67, 1.33, 2])
-        actual = self.index.get_indexer(target)
-        expected = np.array([0, 0, 1, 1], dtype='intp')
-        tm.assert_numpy_array_equal(actual, expected)
+        scalars = [-0.5, 0, 0.5, 1, 1.5, 2, 2.5, 3, 3.5]
+        correct = {'right': {0.5: 0, 1: 0, 2.5: 1, 3: 1},
+                   'left': {0: 0, 0.5: 0, 2: 1, 2.5: 1},
+                   'both': {0: 0, 0.5: 0, 1: 0, 2: 1, 2.5: 1, 3: 1},
+                   'neither': {0.5: 0, 2.5: 1}}
 
-        actual = self.index.get_indexer(target[[0, -1]])
-        expected = np.array([0, 1], dtype='intp')
-        tm.assert_numpy_array_equal(actual, expected)
+        for idx_side in ['right', 'left', 'both', 'neither']:
 
-        target = IntervalIndex.from_breaks([0, 0.33, 0.67, 1], closed='left')
-        actual = self.index.get_indexer(target)
-        expected = np.array([0, 0, 0], dtype='intp')
-        tm.assert_numpy_array_equal(actual, expected)
+            idx = IntervalIndex.from_tuples([(0, 1), (2, 3)], closed=idx_side)
 
-    def test_contains(self):
-        # Only endpoints are valid.
-        i = IntervalIndex.from_arrays([0, 1], [1, 2])
+            for scalar in scalars:
 
-        # Invalid
-        assert 0 not in i
-        assert 1 not in i
-        assert 2 not in i
-
-        # Valid
-        assert Interval(0, 1) in i
-        assert Interval(0, 2) in i
-        assert Interval(0, 0.5) in i
-        assert Interval(3, 5) not in i
-        assert Interval(-1, 0, closed='left') not in i
-
-    def testcontains(self):
-        # can select values that are IN the range of a value
-        i = IntervalIndex.from_arrays([0, 1], [1, 2])
-
-        assert i.contains(0.1)
-        assert i.contains(0.5)
-        assert i.contains(1)
-        assert i.contains(Interval(0, 1))
-        assert i.contains(Interval(0, 2))
-
-        # these overlaps completely
-        assert i.contains(Interval(0, 3))
-        assert i.contains(Interval(1, 3))
+                if scalar in correct[idx_side].keys():
+                    assert idx.get_loc(scalar) == correct[idx_side][scalar]
```

Comment:
I don't understand this one. What is the difference between the two kinds of scalars?

Reply:
Would you elaborate on your question just a little for me? I'm not sure I understand. =/

Reply:
Well, simply, I don't understand what this is testing, or why some scalars should pass and some should fail? 
So can you elaborate on the 'logic'? (which behavioural rule is being followed here?)